app/services/bots/create_service.rb

```diff
@@ -10,11 +10,12 @@ def initialize(user = nil, namespace = nil, bot_type = nil, params = {}) # ruboc
       super(user, params)
       @namespace = namespace
 
+      current_count = @namespace.namespace_bots.count
       user_params = {
-        email: "#{namespace.type.downcase}_#{namespace.puid.downcase}_bot_#{SecureRandom.hex(5)}@iridanext.com",
+        email: "#{namespace.puid}_BOT_#{format('%03d', (current_count + 1))}@iridanext.com",
```

Comment:
**note:** I guess rails is downcasing the emails, so I guess it wasn't as important to uppercase it here.

Reply:
Yeah rails automatically does that. Updated in d55982d7e963f7e8365a2f47f03f3b8060ec9230